jquery.flot.touchNavigate.js

```diff
@@ -72,31 +74,90 @@ can set the default in the options. */
             }
         }
 
+        function isPinchEvent(e) {
+            return e.touches && e.touches.length === 2;
+        }
+
         function onDragStart(e) {
             e.stopPropagation();
             e.preventDefault();
-            plot.getPlaceholder().css('cursor', plot.getOptions().pan.cursor);
-            startPageX = e.touches[0].pageX || e.touches[0].clientX;
-            startPageY = e.touches[0].pageY || e.touches[0].clientY;
-            plotState = plot.navigationState();
+            scaling = isPinchEvent(e);
+
+            if (!isPinchEvent(e)) {
+                scaling = false;
 
-            saveNavigationData(plot, e);
+                plot.getPlaceholder().css('cursor', plot.getOptions().pan.cursor);
+                startPageX = e.touches[0].pageX || e.touches[0].clientX;
+                startPageY = e.touches[0].pageY || e.touches[0].clientY;
+                plotState = plot.navigationState();
+
+                saveNavigationData(plot, e);
+            } else {
+                scaling = true;
+                prevDist = pinchDistance(e);
+            }
         }
 
         function onDrag(e) {
             e.stopPropagation();
             e.preventDefault();
 
-            plot.smartPan({
-                x: startPageX - (e.touches[0].pageX || e.touches[0].clientX),
-                y: startPageY - (e.touches[0].pageY || e.touches[0].clientY)
-            }, plotState);
-
-            saveNavigationData(plot, e);
+            scaling = isPinchEvent(e);
+
+            if (!scaling) {
+                return false;
+                plot.smartPan({
+                    x: startPageX - (e.touches[0].pageX || e.touches[0].clientX),
+                    y: startPageY - (e.touches[0].pageY || e.touches[0].clientY)
+                }, plotState);
+
+                saveNavigationData(plot, e);
+            } else if (isPinchEvent(e)) {
+                var dist = pinchDistance(e);
+                if (Math.abs(100 * (dist - prevDist) / prevDist) > 5) {
```

Comment:
The amount is computed here and inside the onZoomPinch as well. You should refactor this piece of code and the onZoomPinch function to perform a single distance measurement.

Is 5 a magic number?